### ISSUE TITLE
[image_picker] - Request multiple images at once

### ISSUE THREAD

**Issue by acidjazz** [flutter/flutter#16054]
Right now the picker defaults to single mode, it would be nice to be able to have a multi-mode, by passing a parameter when you launch it, then an array is returned of files instead of just one.

for android this might help speed things up
unique code
```java
intent.putExtra(Intent.EXTRA_ALLOW_MULTIPLE, true);
```
full example
```java
Intent intent = new Intent();
intent.setType("image/*");
intent.putExtra(Intent.EXTRA_ALLOW_MULTIPLE, true);
intent.setAction(Intent.ACTION_GET_CONTENT);
startActivityForResult(Intent.createChooser(intent,"Select Picture"), 1);
```

for iOS I think maybe using this library might help, and seems up to date.

https://github.com/hackiftekhar/IQMediaPickerController

**Comment by goodis0** [comment]
would love to see this functionality also

**Comment by mrdimaan** [comment]
this functionality is needed here too.

**Comment by zoechi** [comment]
There are several plugins available that seem to provide that functionality

https://pub.dartlang.org/packages?q=image+picker

- https://pub.dartlang.org/packages/multi_image_picker
- https://pub.dartlang.org/packages/flutter_multiple_image_picker
- https://pub.dartlang.org/packages/custom_multi_image_picker
- https://pub.dartlang.org/packages/multiple_image_picker
...

**Comment by cyanglaz** [comment]
related @https://github.com/flutter/flutter/issues/16054

**Comment by cyanglaz** [comment]
It seems not natively supported by iOS but android seems to support it (>18). If anyone can create a PR for Android, I am happy to review it!

**Comment by kleinpetr** [comment]
@acidjazz I like your suggestion can you create PR? Thanks :)

**Comment by kleinpetr** [comment]
this plugin solved all my problems

https://pub.dev/packages/file_picker

**Comment by cyanglaz** [comment]
I am not able to actively work on this recently. I will un-assign myself from this issue. Feel free to pick this up if anyone is interested!

**Comment by lifenautjoe** [comment]
I've tested everything I could find to do this.

* https://pub.dev/packages/photo_manager 
* https://pub.dev/packages/photo
* https://pub.dartlang.org/packages/multi_image_picker
* https://pub.dartlang.org/packages/flutter_multiple_image_picker
* https://pub.dartlang.org/packages/custom_multi_image_picker
* https://pub.dartlang.org/packages/multiple_image_picker

Packages are either incomplete (image only, no video) or critically buggy and not actively maintained.

iOS and Android both natively support multi-media picking, is it really that hard to support it as part of the official media picking library?

![android-support](https://user-images.githubusercontent.com/12737509/63847283-c7baf180-c98d-11e9-877a-51664775c880.gif)

**Comment by cyanglaz** [comment]
@lifenautjoe Android does natively support a multi image picking UI. However, I am not aware that iOS has a native image picking UI. iOS does have native APIs to query photo assets but it doesn't have any UI binding to it. Ideally, we can expose these APIs so user can directly access photo assets from dart, we can also create a custom UI to pick multiple images on iOS.

**Comment by b-jan** [comment]
![image](https://user-images.githubusercontent.com/24225949/63902006-7e0eed00-ca07-11e9-9cd2-f76a8d72af94.png)

This would be the best UI. 
Same as on this package: https://github.com/Sh1d0w/multi_image_picker

**Comment by Tokenyet** [comment]
@b-jan I agreed, that package is good. However, It required the minSdk to 19 instead of 16, and might need to tweak some [configuration](https://sh1d0w.github.io/multi_image_picker/#/androudtroubleshooting). If the plugin used 16, It would be the best imo.

**Comment by mzimbres** [comment]
I would love to see this feature, any chance of seeing this implemented soon? The lack of this feature degrades usability and is a nightmare to users to have to click the pick image button n times. It would otherwise be painful for developers to rely on unofficial plugins for this basic functionality.

**Comment by fzyzcjy** [comment]
Half a year has passed... Any progress? Thanks

**Comment by v1z4** [comment]
Would love to see it released

**Comment by kw2019ltd** [comment]
any update

**Comment by v1z4** [comment]
I switched to https://pub.dev/packages/file_picker

**Comment by AmG2000** [comment]
Any progress yet? This is a feature that shoud be officialy supported

**Comment by rmControls** [comment]
Any Progress on this... This is the very important feature to have

**Comment by reverie-ss** [comment]
In flutter, every now and then I get a basic requirement.
I start exploring and end up with an 2/3 year old open issue :(

**Comment by ItsPierre** [comment]
I just stumbled over this issue because I need to let the user pick multiple images, too.
I tried many other packages, but they did not satisfy me.
As I further searched for it, I just found [this pull request](https://github.com/flutter/plugins/pull/3783) updated a few weeks ago.
This seems to solve this issue as soon as it is ready to be published in a next update.

I tried the pull request in my code and it seems to work fine.

**Comment by chavesgu** [comment]
please try [images_picker](https://pub.dev/packages/images_picker)

**Comment by Urkman** [comment]
@chavesgu Looks promising, but gives me errors on iOs :(

**Comment by b3nni97** [comment]
Are there any updates?

**Comment by MalteMagnussen** [comment]
Just tried implementing a cool image feature in my Flutter Web App, and I stumbled on this issue. 

Would LOVE to be able to let my users select multiple images at once, instead of having to select one at a time. 

Hope we can get this implemented soon-ish. The PR looks promising. Let's hope we get this soon! <3 

https://github.com/flutter/plugins/pull/3783

**Comment by LongZhai** [comment]
why is this ticket closed? the issue is not fixed yet

**Comment by cyanglaz** [comment]
@LongZhai Just published 0.8.1 which fixes this issue :)

**Comment by LongZhai** [comment]
@cyanglaz Thanks for your prompt response. Could you let me know which function will be used to pick up multiple images and what is the return type? I did not find the function specs.

**Comment by reverie-ss** [comment]
> @cyanglaz Thanks for your prompt response. Could you let me know which function will be used to pick up multiple images and what is the return type? I did not find the function specs.

`getMultiImage()`

**Comment by ydag** [comment]
> @cyanglaz Thanks for your prompt response. Could you let me know which function will be used to pick up multiple images and what is the return type? I did not find the function specs.

Hi @LongZhai,

I created a [PR](https://github.com/flutter/flutter/issues/16054) to update the example app to support the multi-image feature and its implementation. Feel free to check it :)

**Comment by MalteMagnussen** [comment]
Does this Feature work on Flutter Web? 

I tried here: https://github.com/MalteMagnussen/flutterhobby/blob/main/lib/ImageSorting/image_sorter.dart 

The idea is that the user can upload a bunch of images and then sort through them on the site: https://maltemagnussen.com/#/imagesorting and then delete all the ones that weren't dragged to the pile on the right side. 

Sadly you can't use File to delete images on Web afaik, so the feature is kinda on pause for now. 

_This is just a hobby project to practice Flutter and try things like using File and Draggable etc. The idea for the feature came from my brother who takes 300 images per day on Vacation, and wants an easy way to view and sort them._

**Comment by ydag** [comment]
> Does this Feature work on Flutter Web?
> 
> I tried here: https://github.com/MalteMagnussen/flutterhobby/blob/main/lib/ImageSorting/image_sorter.dart#L41
> 
> The idea is that the user can upload a bunch of images and then sort through them on the site: https://maltemagnussen.com/#/imagesorting and then delete all the ones that weren't dragged to the pile on the right side.
> 
> Sadly you can't use File to delete images on Web afaik, so the feature is kinda on pause for now.
> 
> _This is just a hobby project to practice Flutter and try things like using File and Draggable etc. The idea for the feature came from my brother who takes 300 images per day on Vacation, and wants an easy way to view and sort them._
> 
> EDIT: Nevermind, it is Unimplemented.
> 
> `Error: UnimplementedError: pickMultiImage() has not been implemented. at Object.throw_ [as throw] (http://localhost:55539/dart_sdk.js:5059:11) at image_picker_for_web.ImagePickerPlugin.new.pickMultiImage (http://localhost:55539/packages/image_picker_platform_interface/src/method_channel/method_channel_image_picker.dart.lib.js:75:17) at image_picker.ImagePicker.new.getMultiImage (http://localhost:55539/packages/image_picker/image_picker.dart.lib.js:45:48) at image_sorter._SortImageWidgetController.new.getImage (http://localhost:55539/packages/flutterhobby/ImageSorting/image_sorter.dart.lib.js:735:46) at getImage.next (<anonymous>) `
> 
> Will have to wait until this MultiImagePicker works on Web, and you can somehow either delete user images, or zip the kept images and allow user to download that.

Hi @MalteMagnussen,

Unfortunately, this feature is implemented only for Android 4.3+ and iOS 14+ for now. If you have ideas on how to update the web part feel free to share :)

**Comment by yasinarik** [comment]
> @LongZhai Just published 0.8.1 which fixes this issue :)

Hello! I have seen your message but couldn't understand which plugin you were referring. Then I found it. If anybody else wondering, here is the repo:

[Flutter Plugins --> ImagePicker](https://github.com/cyanglaz/plugins/blob/master/packages/image_picker/image_picker/lib/image_picker.dart)

### Very Important Feature
I cannot see any property **limiting** the image count that can be selected. Will there be that feature? It is very important I think because lots of developers want to control how many images a user can pick with multi-picking.

Could you please implement that too? 

If anybody wondering, currently, I am using this package. However, it doesn't have a feature to limit the maximum pickable image count. :(
[File Picker v3.0.3](https://github.com/miguelpruivo/flutter_file_picker)

Using this codepiece
```dart
await FilePicker.platform.pickFiles(type: FileType.image, allowMultiple: true);
```

**Comment by ydag** [comment]
Hi @yasinarik,

It would be great if you open a new issue for this proposal since this issue is closed and it can be hard for people to notice your comment. Also, you can explain a bit more why this feature is important and what are use-cases. 

Thanks!
